src/network/networkd-link-bus.c

```diff
@@ -10,12 +11,36 @@
 static BUS_DEFINE_PROPERTY_GET_ENUM(property_get_operational_state, link_operstate, LinkOperationalState);
 static BUS_DEFINE_PROPERTY_GET_ENUM(property_get_administrative_state, link_state, LinkState);
 
+static int bus_link_method_dhcp_send_renew(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        Link *link = userdata;
+        int r;
+
+        assert(link);
+
+        if (!link->network)
+                return sd_bus_error_set(error, BUS_ERROR_UNMANAGED_INTERFACE, "Interface is not managed by systemd-networkd");
+
+        if (!FLAGS_SET(link->network->dhcp, ADDRESS_FAMILY_IPV4))
+                return sd_bus_error_set(error, BUS_ERROR_DHCP4_CLIENT, "DHCPv4 client is not enabled on the interface");
+
+        if (!link->dhcp_client)
+                return sd_bus_error_set(error, BUS_ERROR_DHCP4_CLIENT, "DHCPv4 client is not configured on the interface");
+
+        r = sd_dhcp_client_send_renew(link->dhcp_client);
+        if (r < 0)
+                return r;
+
+        return sd_bus_reply_method_return(message, NULL);
+}
+
 const sd_bus_vtable link_vtable[] = {
         SD_BUS_VTABLE_START(0),
 
         SD_BUS_PROPERTY("OperationalState", "s", property_get_operational_state, offsetof(Link, operstate), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
         SD_BUS_PROPERTY("AdministrativeState", "s", property_get_administrative_state, offsetof(Link, state), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
 
+        SD_BUS_METHOD("DHCPv4SendRenew", NULL, NULL, bus_link_method_dhcp_send_renew, 0),
```

Comment:
hmm, wouldn't a more generic method "Renew()" be better? i.e. something that is less specific about it does, but generically renews all addresses it might have acquired dynamically, regardless which protocol is used?

Reply:
Sounds good. Moreover, it may be good to optionally take `as` argument to specify protocols to renew.